docs/testing/README.md

```diff
@@ -0,0 +1,26 @@
+# Testing in LocalStack
+
+- [Integration Tests](integration-tests/README.md)
+- [Parity Testing](parity-testing/README.md)
+- [Multi-account and Multi-region Testing](multi-account-region-testing/README.md)
+- [Terraform Tests](terraform-tests/README.md)
+
+## Rules for stable tests
+
+Through experience, we encountered some guiding principles and rules when it comes to testing LocalStack.
+These aim to ensure a stable pipeline and keep flakes minimal. Any newly added test and feature should keep these in mind!
+
+| **ID** | **Rule** |
+|-------|-----------|
+| R01   | Inform code owners and/or test authors when a test is marked as flaky so that it can be fixed ASAP. |
+| R02   | Do not assume external dependencies (AWS resources, files, packages, images, licenses) are indefinitely available. |
+| R02a  | Where possible, add automatic creation of dependencies if they no longer exist (e.g., S3 buckets, roles). |
```

Comment:
R02 are common issues with our test pipeline, but we don't really have a clear solution yet.
And I am not really sure what R02a means exactly to be honest...

Reply:
In a failing test involving pods, it failed because some S3 bucket that got accessed was deleted. I'm not sure if it was a write or a read access.
But generally, if there is some test where something should be written to a bucket `x`, you can add a check beforehand if `x` exists, and if it doesn't just create it before using it.

Reply:
Also, I think manifesting mindfulness of these issues in the rules will lead to development of best practices for these problems in the future